exercises.ipynb

```diff
@@ -172,22 +319,230 @@
     "\n",
     "HINT: Write a separate function to get the current exchange rates! That can be reused!"
    ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 152,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import pandas as pd"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 153,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_currencies(request_url):\n",
+    "    \n",
+    "    response = requests.get(request_url)\n",
+    "    if response.status_code == 200:\n",
+    "        rates = json.loads(response.text)[\"rates\"]\n",
+    "    return rates"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 159,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def load_from_file(input_file, output_file=\"assets_new.csv\", base_curr = \"EUR\"):\n",
+    "    assets_df = pd.read_csv(input_file)\n",
+    "    rates = get_currencies(f\"https://api.exchangeratesapi.io/latest?base={base_curr}\")\n",
+    "    rates_df = pd.DataFrame.from_dict(rates, orient=\"index\")\n",
+    "    assets_new_df = pd.merge(assets_df,\n",
+    "                         rates_df,\n",
+    "                         how=\"left\",\n",
+    "                         left_on=\"curr\",\n",
+    "                         right_index=True)\n",
+    "    assets_new_df.columns = [\"value\", \"curr\", \"exchange_rate\"]\n",
+    "    assets_new_df[base_curr] = assets_new_df[\"value\"] / assets_new_df[\"exchange_rate\"]\n",
+    "    assets_new_df.drop(columns=\"exchange_rate\", inplace=True)\n",
+    "    assets_new_df.to_csv(output_file, compression='gzip', index=False)\n"
```

Comment:
Not sure why you chose to store this in an intermediate file, if you're worried about the API call just cache it! Less state, less places for things to go wrong.